pages/api/cck/route.ts

```diff
@@ -1,5 +1,49 @@
 import type { NextApiRequest, NextApiResponse } from 'next';
 import prisma from '@/components/db-connection-prisma';
+import type { VolunteerApiResponse } from '@/components/api-helpers';
+import { makeResponse } from '@/components/api-helpers';
+import { RouteDeliveryResponse } from '@/components/db-connection/RouteDelivery';
+import { EventCategoryResponse } from '@/components/db-connection/EventCategory';
+import { EventResponse } from '@/components/db-connection/Event';
+import { RouteResponse } from '@/components/db-connection/Route';
+
+const eventCategoryRef = 'meal-prep-delivery';
+
+const eventRoleRef = 'delivery';
+
+const errorResponses: { [key: string]: VolunteerApiResponse; } = Object.freeze({
+    notFound: {
+        code: 404,
+        message: "Route not found"
+    },
+    invalidPasscode: {
+        code: 403,
+        message: "Passcode is invalid"
+    },
+    noPasscode: {
+        code: 403,
+        message: "For now, you MUST provide a passcode to access a route."
+    }
+});
+
+/**
+ * Parses a date string into multiple date formats
+ */
+function getDateFormats(dateString: string): [string, string] {
```

Comment:
are we not able to use Javascript Date objects here? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date

Reply:
We can, but I'm unaware of the Javascript Date object's ability to parse date strings in a multitude of formats.  There are 3rd party libraries, however, that can simplify this process. This was a relatively simple ask, though, so I chose to just do the parsing manually.

https://stackoverflow.com/questions/10430321/how-to-parse-a-dd-mm-yyyy-or-dd-mm-yyyy-or-dd-mmm-yyyy-formatted-date-stri

Reply:
I have, however, simplified the parsing using the Javascript string's split method.